src/lib/transaction_logic/mina_transaction_logic.ml

```diff
@@ -1612,9 +1612,9 @@ module Make (L : Ledger_intf.S) :
 
         let if_ = value_if
 
-        let is_pos (t : t) = Sgn.equal t.sgn Pos
+        let is_non_negative (t : t) = Sgn.equal t.sgn Pos
 
-        let is_neg (t : t) = Sgn.equal t.sgn Neg
+        let is_non_positive (t : t) = Sgn.equal t.sgn Neg
```

Comment:
This renaming is wrong, revert

Reply:
It returns `true` for `{ magnitude : 0; sgn : Neg }`, which behaves like a proper `0`

Reply:
Reverted. It would have been very helpful to know in advance the invariant that `{magnitude = zero; sgn = Neg}` is never expected to be produced (as both these functions break down on this value. Sadly, the type's internal structure is exposed, so it's still possible to create such a value by hand, so technically it's not guaranteed to hold. I added a comment to explain this.